### PR TITLE
fix bug in weixin min program and add support to ali min program

### PR DESCRIPTION
**Pull request by SyMind**
issue #892

### CODE REVIEW

**Comment by mcollina** [comment]
Can you please restore the test, and add a similar one for ali?

**Comment by SyMind** [comment]
@mcollina I don't think i can do it. Because weixin and ali min program is a platform that Tencent and Alibaba does not open source, and they donot have tools for unit testing. 

I read code in https://github.com/mqttjs/MQTT.js/blob/master/test/browser/wx.js, but i have not find ../helpers/wx. I wonder guy who add weixin min program support how to do it.

File: lib/connect/ws.js [hunk 1]
```diff
@@ -82,7 +82,9 @@ function buildBuilderBrowser (client, opts) {
       opts.port = parsed.port
     }
   }
-  return createWebSocket(client, opts)
+  var ws = createWebSocket(client, opts)
+  console.log('ws -> ', ws)
```
Comment (mcollina) [inline lib/connect/ws.js]: Can you remove this console.log?

**Comment by SyMind** [comment]
@taoqf i need your help.

File: lib/connect/index.js [hunk 2]
```diff
@@ -87,6 +90,7 @@ function connect (brokerUrl, opts) {
           case 'wx':
             opts.protocol = 'wxs'
             break
+          case 'alis':
```
Comment (mcollina) [inline lib/connect/index.js]: I fear this might throw for alis.

Comment (SyMind) [reply lib/connect/index.js]: case 'ali':
            opts.protocol = 'alis'
            break

File: lib/connect/ws.js (hunk 1 above)
Comment (SyMind) [reply lib/connect/ws.js]: yes

**Comment by taoqf** [comment]
@SyMind I might use that, too. I am glad to see this pr. 
@mcollina , Actually I don't think anyone could give a unit test for ali or wx, it could only run in it's [devetools](https://developers.weixin.qq.com/miniprogram/dev/devtools/devtools.html) and in [weixin](https://weixin.qq.com/) or [alipay](https://www.alipay.com/).

**Comment by mcollina** [comment]
@SyMind is this ready to go?

@taoqf can you review as well?

**Pull request merged**